haystack/components/routers/zero_shot_text_router.py

```diff
@@ -0,0 +1,159 @@
+            doing a softmax of the entailment score vs. the contradiction score.
+        :param model: The name or path of a Hugging Face model for zero-shot text classification.
+        :param device: The device on which the model is loaded. If `None`, the default device is automatically
+            selected. If a device/device map is specified in `pipeline_kwargs`, it overrides this parameter.
+        :param token: The API token used to download private models from Hugging Face.
+            If this parameter is set to `True`, the token generated when running
+            `transformers-cli login` (stored in ~/.huggingface) is used.
+        :param pipeline_kwargs: Dictionary containing keyword arguments used to initialize the
+            Hugging Face pipeline for zero shot text classification.
+        """
+        torch_and_transformers_import.check()
+
+        self.token = token
+        self.labels = labels
+        self.multi_label = multi_label
+        component.set_output_types(self, **{label: str for label in labels})
+
+        pipeline_kwargs = resolve_hf_pipeline_kwargs(
+            huggingface_pipeline_kwargs=pipeline_kwargs or {},
+            model=model,
+            task="zero-shot-classification",
+            supported_tasks=["zero-shot-classification"],
+            device=device,
+            token=token,
+        )
+        self.pipeline_kwargs = pipeline_kwargs
+        self.pipeline = None
+
+    def _get_telemetry_data(self) -> Dict[str, Any]:
+        """
+        Data that is sent to Posthog for usage analytics.
+        """
+        if isinstance(self.pipeline_kwargs["model"], str):
+            return {"model": self.pipeline_kwargs["model"]}
+        return {"model": f"[object of type {type(self.pipeline_kwargs['model'])}]"}
+
+    def warm_up(self):
+        if self.pipeline is None:
+            self.pipeline = pipeline(**self.pipeline_kwargs)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+        """
+        serialization_dict = default_to_dict(
+            self,
+            labels=self.labels,
+            pipeline_kwargs=self.pipeline_kwargs,
+            token=self.token.to_dict() if self.token else None,
+        )
+
+        pipeline_kwargs = serialization_dict["init_parameters"]["pipeline_kwargs"]
+        pipeline_kwargs.pop("token", None)
+
+        serialize_hf_model_kwargs(pipeline_kwargs)
+        return serialization_dict
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "TransformersZeroShotTextRouter":
+        """
+        Deserialize this component from a dictionary.
+        """
+        deserialize_secrets_inplace(data["init_parameters"], keys=["token"])
+        deserialize_hf_model_kwargs(data["init_parameters"]["pipeline_kwargs"])
+        return default_from_dict(cls, data)
+
+    def run(self, text: str) -> Dict[str, str]:
```

Comment:
@component.output_types decorator ?

Reply:
great point, just for my own knowledge why does it work without it?

Reply:
Not 100% sure, but I'd say it works because of the manual output setup you do in the constructor. The output keys are dynamic - known at runtime when you initialize the component with the labels.